services/src/util/middleware.rs

```diff
@@ -0,0 +1,121 @@
+use std::future::{ready, Ready};
+
+use actix_web::{
+    dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform},
+    http::header::{HeaderName, HeaderValue},
+    Error, HttpMessage,
+};
+use futures_util::future::LocalBoxFuture;
+use tracing::{event_enabled, Level};
+use tracing_actix_web::RequestId;
+
+const REQUEST_ID_HEADER: &str = "x-request-id";
+
+// There are two steps in middleware processing.
```

Comment:
briefly describe what the Transform does here, instead of explaining the middleware processing.

Reply:
[cc147b9](https://github.com/geo-engine/geoengine/pull/955/commits/cc147b9b31152d18b5705bf715d96e290064cc81)